### PR TITLE
Gencred: require --context; do not default to current-context

### PR DESCRIPTION
**Pull request by clarketm**

### CODE REVIEW

File: gencred/README.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ The following is a list of supported options for the `gencred` CLI. All options
 
 ```console
   -c, --certificate      Authorize with a client certificate and key.
-      --context string   The name of the kubeconfig context to use.
+      --context string   The name of the kubeconfig context to use. (default current-context)
```
Comment (fejta) [inline gencred/README.md]: Using the current context is a footgun, can we just remove it?

Comment (clarketm) [reply gencred/README.md]: At the expense of additional verbosity in some cases, yes, we can definitely make the context a *required* options.

Comment (fejta) [reply gencred/README.md]: I'm more concerned about potentially creating admin credentials for the wrong cluster than avoiding the hassle and typing of determining the correct context.

Comment (clarketm) [reply gencred/README.md]: Done and added unit test(s) to verify.

**Pull request merged**